doboto/Endpoint.py

```diff
@@ -80,3 +80,51 @@ def pages(self, request_url, expect, params=None):
                 next_url = None
 
         return items
+
+    def action_result(self, action, wait, poll, timeout):
+        """
+        General action result processor for waiting
+        """
+
+        start_time = time.time()
+
+        while wait and action["status"] == "in-progress":
+
+            time.sleep(poll)
+            try:
+                action = self.do.action.info(action["id"])
+            except:
+                pass
```

Comment:
Ya, that's the idea. It's just polling to see if the state changed.  If there's a network blip, I didn't want it to affect anything.